### PR TITLE
PagerDuty APIv2 ServiceKey to RoutingKey

### PR DESCRIPTION
**Pull request by onlynone**
###### Required for all non-trivial PRs
- [x] Rebased/mergable
- [x] Tests pass
- [x] CHANGELOG.md updated
- [x] Sign [CLA](https://influxdata.com/community/cla/) (if not already signed)

I think this should address issue #1938

### CODE REVIEW

**Comment by nathanielc** [comment]
@onlynone Thanks this is great! 

@russorat Can you validate that this works for you?

**Comment by russorat** [comment]
@nathanielc we will need to update chronograf alert rule builder for this change.

**Comment by onlynone** [comment]
@russorat I'm not sure where that is. Would you be able to push a change for that, or point me where to look? Is there test coverage for the builder?

**Comment by russorat** [comment]
@onlynone @nathanielc is there any way for the chronograf team to know when to use serviceKey vs routingKey when generating tickscripts via the alert rule builder?

**Comment by nathanielc** [comment]
@russorat Yes, serviceKey is v1 routingKey is v2.

File: pipeline/alert.go [hunk 1]
```diff
@@ -1016,9 +1016,9 @@ func (n *AlertNodeData) PagerDuty2() *PagerDuty2Handler {
 type PagerDuty2Handler struct {
 	*AlertNodeData `json:"-"`
 
-	// The service key to use for the alert.
+	// The routing key to use for the alert.
 	// Defaults to the value in the configuration if empty.
-	ServiceKey string `json:"serviceKey"`
+	RoutingKey string `json:"routingKey"`
 }
```
Comment (nathanielc) [inline pipeline/alert.go]: If you add a `ServiceKey` function to the `PagerDuty2Handler`  that takes a string and sets the `RoutingKey` field then we maintain backwards compatibility.

**Comment by onlynone** [comment]
@nathanielc I tried making the changes you recommended, but I'm getting the error `no method or property "routingKey" on *pipeline.PagerDuty2Handler`. This is the diff I came up with based on your feedback and advice:

```
diff --git a/pipeline/alert.go b/pipeline/alert.go
index 90000f07..72ce5101 100644
--- a/pipeline/alert.go
+++ b/pipeline/alert.go
@@ -1018,7 +1018,14 @@ type PagerDuty2Handler struct {
 
        // The routing key to use for the alert.
        // Defaults to the value in the configuration if empty.
-       RoutingKey string `json:"routingKey"`
+       RoutingKey string `tick:"ServiceKey" json:"routingKey"`
+}
+
+// Allow ServiceKey as backwards compatible way to set the routing key
+// tick:property
+func (pd2 *PagerDuty2Handler) ServiceKey(serviceKey string) *PagerDuty2Handler {
+       pd2.RoutingKey = serviceKey
+       return pd2
 }
 
```

I'm far from knowledgeable about how all the structs/properties/functions/tick script specification interact. If there's something I'm missing that would allow `alert().pagerduty2().routingKey('blah')` and `alert().pagerduty2().serviceKey('blah')` to both work and just update the `pipeline.PagerDuty2Handler.RoutingKey` property. Feel free to update this PR (I think maintainers have modification access to this PR).

Would it not make more sense to just treat `pagerduty2().serviceKey()` as a typo of `pagerduty2().routingKey()` ? In which case it would just be a bug fix, not a feature of the API that needs to be supported forever? All the documentation just mentions `.routingKey()`. If it had been `.routingKee()` it would just be fixed without keeping the misspelling, right?

**Comment by onlynone** [comment]
Hmm, I think if I do this it works:

```
diff --git a/pipeline/tick/alert.go b/pipeline/tick/alert.go
index 35825c85..7067dfbc 100644
--- a/pipeline/tick/alert.go
+++ b/pipeline/tick/alert.go
@@ -120,7 +120,8 @@ func (n *AlertNode) Build(a *pipeline.AlertNode) (ast.Node, error) {
 
 	for _, h := range a.PagerDuty2Handlers {
 		n.Dot("pagerDuty2").
-			Dot("routingKey", h.RoutingKey)
+			Dot("routingKey", h.RoutingKey).
+			Dot("serviceKey", h.ServiceKey)
 	}
 
 	for _, h := range a.PushoverHandlers {
diff --git a/pipeline/alert.go b/pipeline/alert.go
index 90000f07..358f2ee9 100644
--- a/pipeline/alert.go
+++ b/pipeline/alert.go
@@ -1019,6 +1019,16 @@ type PagerDuty2Handler struct {
 	// The routing key to use for the alert.
 	// Defaults to the value in the configuration if empty.
 	RoutingKey string `json:"routingKey"`
+
+	// tick:ignore
+	_ string `tick:"ServiceKey"`
+}
+
+// Allow ServiceKey as backwards compatible way to set the routing key
+// tick:property
+func (pd2 *PagerDuty2Handler) ServiceKey(serviceKey string) *PagerDuty2Handler {
+	pd2.RoutingKey = serviceKey
+	return pd2
 }
 
 // Send the alert to HipChat.
```

Does that make sense?

**Comment by onlynone** [comment]
@nathanielc I think this should allow `.serviceKey` for backwards compatibility.

**Comment by onlynone** [comment]
@nathanielc @russorat Do these changes look good? I'd like to get this change merged because I've got another PR I'd like to open that involves pagerduty2, but it would create conflicts with this branch if I based it off `master`. Right now I've got the other branch on top of this one, so I don't want to open it until this one is resolved somehow.

**Comment by onlynone** [comment]
@russorat let me know if there's anything else I can do for this PR.

**Comment by onlynone** [comment]
@nathanielc @russorat ping?

**Comment by russorat** [comment]
this is fine for chronograf. our alert rule builder doesn't let you override the routingKey in the tickscript anyway. and as long as the api for setting the config is backwards compatible, we are good.

**Comment by onlynone** [comment]
@nathanielc good to merge?

**Comment by onlynone** [comment]
@russorat @nathanielc is there someone else who needs to sign off on this?

**Comment by onlynone** [comment]
@stevebang @russorat @nathanielc it's been 2 months since I pushed the last commit on this PR, which russ and nathaniel said looked good, is there anything else I need to do to get it merged?

**Comment by onlynone** [comment]
@timhallinflux are there any more steps required to merge this into master now that it's been added to the 1.5.1 milestone?

**Pull request merged**